java/org/apache/coyote/http11/Http11InputBuffer.java

```diff
@@ -841,7 +846,8 @@ private HeaderParseStatus parseHeader() throws IOException {
 
             if (chr == Constants.CR && prevChr != Constants.CR) {
```

Comment:
This appears to essentially ignore (all?) CR characters in headers. I'm not sure that's what we want.

Reply:
It shouldn't. What it should do is detect if a line starts with CRLF or LF and if it does set the HeaderParseStatus.DONE state. Anything else is passed through to be parsed as the start of the header name. That particular line is handling the case where the lines starts with CR which may be the start of a CRLF sequence.